### DISCUSSION TITLE
Question about whether tauri will support webserver

### DISCUSSION THREAD

**Discussion by hanxiaohuivc** [tauri-apps/tauri#2537] [Q&A] [answered]
Hello everyone. I want to ask a question about tauri architecture. For example, I have a vuejs project that requests data from webservers through js. Now I can use tauri to package vuejs as a desktop and also access webservers. My question is, will tauri provide some way to package webservers to desktop?
Thank you.

**Comment by tcastelly** [comment, marked as answer]
Hello,

Tauri is a framework to embed Javascript web application in native "desktop" application. Rust is used as system language to communicate with your file system.

Maybe I don't understand correctly your project, please fix me if I'm wrong.
If you have a VueJS application with a REST api (or something else) to communicate with your backend. You can use Tauri to create a native application with your existing VueJS and continue to call your backend using the same queries.

"My question is, will tauri provide some way to package webservers to desktop?"
If by `webservers` you mean `backend`. It depends of you project. But yes you can release a native application with:
- VueJS as client part
- Rust as backend

If you are familiar with Electron, you can see the differences between both:
https://github.com/tauri-apps/tauri#comparison-between-tauri-and-electron

**Reply by hanxiaohuivc** [reply]
Hello. thank you for your reply. Package vuejs as a desktop, and interact with the backend REST api through the original method. This method is very good, and I have also tried it. What I also want to know is, if I want to package the back-end functions with vuejs as a desktop, do I need to re-implement the backend functions using rust? Or does tauri provide other ways?

**Reply by tcastelly** [reply]
It depends of your back-end. If you are using NodeJS or Python it should be possible to embedded and call them as [sidecard](https://tauri.studio/en/docs/usage/guides/bundler/sidecar/).

I personally have a project with a part of the back-end written in Golang. I use FFI/cgo.[ Here an example.](https://github.com/tcastelly/vue-tauri-vite).

The Tauri team has also planned to use other bindings as: Deno, Go, Nim, Python, C++.

**Reply by hanxiaohuivc** [reply]
Thank you very much！
